runtest.py

```diff
@@ -111,255 +121,181 @@ def kill_legacy_process(self):
-                        test_name = test_case.split(".")[-1]
-                        case_data[test_case]["TEST_SCRIPT_PY_DIR_PATH"] = os.sep.join(test_case.split(".")[:-1])
-                        if os.path.exists(test_case_fp):
+                            test_case_fp = test_case.replace(".", os.sep) + ".py"
+                            test_name = test_case.split(".")[-1]
                             test_case_module_name = test_case
-                            test_env = self.get_test_env(**case_data[test_case])
-                        else:
-                            self.logger.error("Test script [%s] is not exist!" % test_case_fp)
-                            test_env = None
-
-                    # if webdriver is enable, we need to get parameters for running browsers
-                    if self.webdriver['enable']:
-                        test_env['webdriver'] = "True"
-                        if self.webdriver['run_firefox']:
-                            test_env['browser_type'] = "firefox"
-                            data = self.loop_test(test_case_module_name, test_name, test_env)
-                        if self.webdriver['run_chrome']:
-                            test_env['browser_type'] = "chrome"
-                            data = self.loop_test(test_case_module_name, test_name, test_env)
-                    else:
-                        data = self.loop_test(test_case_module_name, test_name, test_env)
-
-                    if self.online:
-                        if self.perfherder_revision:
-                            self.upload_agent_obj.upload_register_data(input_suite_fp, type, self.perfherder_suitename)
-                        self.upload_agent_obj.upload_videos(data)
+                            runtime_case_data["TEST_SCRIPT_PY_DIR_PATH"] = os.sep.join(test_case.split(".")[:-1])
+                            # if webdriver is enable, we need to get parameters for running browsers
+                            if not os.path.exists(test_case_fp):
+                                self.logger.error("Test script [%s] is not exist!" % test_case_fp)
+                                continue
+                            if self.exec_config['user-simulation-tool'] == self.global_config['default-user-simulation-tool-webdriver']:
+                                runtime_case_data['webdriver'] = "True"
+                                for browser_type in self.exec_config['webdriver-run-browser']:
+                                    runtime_case_data['browser_type'] = browser_type
+                                    test_env = self.get_test_env(**runtime_case_data)
+                                    data = self.loop_test(test_case_module_name, test_name, test_env)
+                                    self.case_teardown(data)
+                            else:
+                                test_env = self.get_test_env(**runtime_case_data)
```

Comment:
**TODO**:
Maybe we can set `runtime_case_data['browser_type'` here, so `baseTest.py` can use one line `self.browser_type = os.getenv("browser_type")` to replace L156-160.

Ref: https://github.com/ShakoHo/Hasal/blob/41e534e01336dddee91f4551640e7fcb5cec8ec5/lib/baseTest.py#L156-L160